### PR TITLE
[e2e tests] improvements in the VM pool suite

### PR DESCRIPTION
**Pull request by enp0s3**
Signed-off-by: Igor Bezukh <ibezukh@redhat.com>

**What this PR does / why we need it**:
* Taking the VM pool scale tests out of quarantine - according to Testgrid these tests were passing the periodic lanes for a couple of weeks, hence taking them back as they proved to be stable.

* Improving the time it takes to run some of the VM pool tests -  the VM pool suite contains tests with running VMs. These tests only observe the statuses of the VMIs and the VM pool resources. They don't interact directly with the guest OS. Therefore we can use a diskless approach when a VM runs the "no bootable device" phase. It will save the time it takes to setup the container-disk.

**Special notes for the reviewer**:
* The minimum amount of RAM required to run SeaBIOS is 1M, I've decided to put 2Mi to be on the safe side.
* changes were tested on a local 1.23 based setup, I've ran them w/o the quarantine label, and also using `gmeasure` to
run them in a loop for 100 times in a row.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: tests/pool_test.go [hunk 1]
```diff
@@ -115,7 +115,7 @@ var _ = Describe("[sig-compute]VirtualMachinePool", func() {
 
 	newVirtualMachinePool := func() *poolv1.VirtualMachinePool {
 		By("Create a new VirtualMachinePool")
-		pool := newPoolFromVMI(libvmi.NewCirros())
+		pool := newPoolFromVMI(libvmi.New(libvmi.WithResourceMemory("1Mi")))
```
Comment (xpivarc) [inline tests/pool_test.go:118]: I think this is not enough. Depending on the firmware you need 32/132Mi at least.

Comment (enp0s3) [reply tests/pool_test.go:118]: @xpivarc I've been testing it locally and seems like the VM managed to launch and VMI status was running. Would you like me to do some double check with QEMU minimal RAM requirements to launch SeaBIOS?

**Review by xpivarc — COMMENTED** [review]
/approve
As discussed this can now probably run in parallel

File: tests/pool_test.go (hunk 1 above)
Comment (enp0s3) [reply tests/pool_test.go]: Confirmed that minimum RAM required for seaBIOS is 1M, increased to 2Mi to be on the safe side. Please see updated PR  description

Comment (xpivarc) [reply tests/pool_test.go]: Thanks Igor. My numbers were based on the assumption we want to boot the VMs but that is not a goal of these tests.

**Comment by EdDev** [comment]
The [job failed](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8185/pull-kubevirt-e2e-k8s-1.22-sig-compute/1552222991293616128) on the `Update` collision.

I think #8116 should take care of it.

**Review by EdDev — APPROVED** [review]

**Pull request merged**